apps/judicial-system/backend/src/app/modules/case/case.service.ts

```diff
@@ -416,69 +426,46 @@ export class CaseService {
       })
   }
 
-  private async deliverRulingToCourt(theCase: Case): Promise<boolean> {
-    return isIndictmentCase(theCase.type)
-      ? this.deliverIndictmentRulingToCourt(theCase)
-      : this.deliverSignedRulingToCourt(theCase)
-  }
-
-  private async deliverIndictmentCourtRecortToCourt(
+  private async deliverProsecutorDocumentsToCourt(
     theCase: Case,
   ): Promise<boolean> {
-    return this.fileService
-      .getAllCaseFiles(theCase.id)
-      .then((caseFiles) =>
-        caseFiles.find(
-          (caseFile) => caseFile.category === CaseFileCategory.COURT_RECORD,
-        ),
-      )
-      .then(async (caseFile) => {
-        if (!caseFile) {
-          throw new Error(
-            `Failed to find the court record for case ${theCase.id} in the database`,
-          )
-        }
-
-        if (!caseFile.key) {
-          throw new Error(
-            `Missing AWS S3 key for the court record of case ${theCase.id}`,
-          )
-        }
-
-        const title = this.formatMessage(courtUpload.courtRecord, {
-          courtCaseNumber: theCase.courtCaseNumber,
-        })
-
-        return this.awsS3Service
-          .getObject(caseFile.key)
-          .then((buffer) =>
-            this.courtService.createDocument(
-              theCase.id,
-              theCase.courtId,
-              theCase.courtCaseNumber,
-              CourtDocumentFolder.COURT_DOCUMENTS,
-              title,
-              caseFile.name.replace(/^.+\./, `${title}.`),
-              caseFile.type,
-              buffer,
-            ),
-          )
-          .then(() => true)
-      })
-      .catch((reason) => {
-        // Tolerate failure, but log error
-        this.logger.error(
-          `Failed to deliver the court record for case ${theCase.id} to court`,
-          { reason },
-        )
+    return isIndictmentCase(theCase.type)
+      ? (await this.deliverCaseFileToCourt(
```

Comment:
Consider refactoring `deliverCaseFileToCourt` function to accept `CaseFileCategory[]` instead of a single `CaseFileCategory` and then run operations on each of the CaseFileCategory.

Reply:
I intended to do this in a future PR as we are rewriting/refactoring queuing as a whole. But I will take a look.

Reply:
Done, managed to delete some code 🥇